DEVELOPMENT.md

```diff
@@ -429,12 +406,71 @@ git submodule update --init --remote
 git submodule foreach "git fetch --all"
 ```
 
-Then run tests as usual: `cargo test`.
+Then run tests as usual: `cargo test --all-features --all-targets`.
+
+We strive for two kind of tests, the classic unit tests inside of the
+implementation files. Additionally we have a suit of integration tests esp. to
+assert the API the proxy provides from a high-level consumer perspective, these
+can be found under `proxy/tests`. To find out where to place and how to layout
+tests check Rust book [test chapter][rt].
+
+### File structure
+
+We understand the GraphQL API as one potential surface to expose the
+Application domain logic. Therefore we try to treat it as a thin layer exposing
+well-typed entities. The heavy lifting is done in the modules named after the
+protocols we consume - `CoCo` for code collaboration and `Registry` for global
```

Comment:
Links to the two repos could be helpful here for an outside reader.

Reply:
Expanded in 536b402.